core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -791,19 +791,20 @@ public IndexMetaData build() {
                 try {
                     minimumCompatibleLuceneVersion = org.apache.lucene.util.Version.parse(stringLuceneVersion);
                 } catch (ParseException ex) {
-                    throw new IllegalStateException("Cannot parse lucene version [" + stringLuceneVersion + "] in the [" + SETTING_VERSION_MINIMUM_COMPATIBLE +"] setting", ex);
+                    throw new IllegalStateException("Cannot parse lucene version [" + stringLuceneVersion + "] in the [" + SETTING_VERSION_MINIMUM_COMPATIBLE + "] setting", ex);
                 }
             } else {
                 minimumCompatibleLuceneVersion = null;
             }
 
-            return new IndexMetaData(index, version, state, numberOfShards, numberOfReplicas, tmpSettings, mappings.build(),
+            final String uuid = settings.get(SETTING_INDEX_UUID, INDEX_UUID_NA_VALUE);
```

Comment:
I am not sure but shouldn't this one be a generated one if it's not in the settings?